tests/expected_tests.cc

```diff
@@ -0,0 +1,39 @@
+#include "gtest/gtest.h"
+#include "stout/expected.h"
+
+stout::expected<int, std::string> divide(int divident, int divider) {
+  if (!divider) {
+    return stout::make_unexpected("divide by zero");
+  } else {
+    return divident / divider;
+  }
+}
+
+TEST(Expected, SucceedResult) {
+  const auto result = divide(6, 3);
```

Comment:
Generally, minimize use of `auto` unless it's either required, or if it strictly improves readability (e.g. by removing type duplication, e.g. `std::unique_ptr<Foo> foo = std::make_unique<Foo>();` is a great place to declare the LHS type as `auto`): https://google.github.io/styleguide/cppguide.html#Type_deduction

Reading this test alone (without looking at the function definition above), I have no idea what type `result` is. Even if I expect that `result` is a `stout::expected<>`, it's still not clear from this type which value type it wraps (`int`? `int32`? `float`? `double`?). Using an explicit return type here spares the reader from having to reference the function definition above in order to understand this line.